nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryAPI.java

```diff
@@ -4364,16 +4364,20 @@ public void testLocking() {
         assertNotNull(lock);
         assertEquals("Administrator", lock.getOwner());
 
-        // nobody can re-lock without unlock, even Administrator
-        try {
-            session.setLock(docRef);
-            fail();
-        } catch (LockException e) {
-            assertTrue(e.getMessage(), e.getMessage().contains("Document already locked"));
-        }
+        Lock relock = session.setLock(docRef);
```

Comment:
You could add a comment before this line like `// a user can call setLock again on their own lock without error`